index.rst

```diff
@@ -302,107 +305,152 @@ Is this case, we could have something like:
+
+#.  In the configuration repository for the given CSC (e.g `ts_config_attcs <https://github.com/lsst-ts/ts_config_attcs>`_ for the ATDome) there shall be a ``_init.yaml`` file defining all values that are expected to be relatively static in operations (we intentionally use "_init" instead of "_default").
+
+    - See this :download:`example _init.yaml <_static/_init.yaml>` for the ATSpectrograph CSC.
+    - This file is the first configuration file to get loaded by the CSC
+    - It is not possible to load this file using the `configurationToApply` attribute
+    - Note that all CSCs having multiple algorithms, each with different required attributes, must have an initial set of defaults in this file.
+
+#.  Also in the configuration repository for the given CSC, there shall be a file corresponding to each site where the CSC is used (e.g. ``_summit.yaml, _ncsa.yaml, _base.yaml``).
+    These files contain site specific configuration parameters such as IP addresses and ports.
+    SalObj determines what site file should be loaded automatically by parsing the ``LSST_DDS_PARTITION_PREFIX`` environment variable
+
+    - See this :download:`example _summit.yaml <_static/_summit.yaml>` for the ATSpectrograph CSC.
+    - This file is the second configuration file to get loaded by the CSC and can override any previously declared values.
+    - It is not possible to load this file using the `configurationToApply` attribute
+    - Between this file and the ``_init.yaml`` file, **the configuration must be fully defined**.
 
-#.  On the configuration repository there shall be a ``_base.yaml`` file defining all the base configuration values (we use "base" instead of "default").
+#.  An additional configuration file provides overrides for the configuration parameters set by the previous files.
 
-    - See this :download:`example _base.yaml <_static/_base.yaml>` for the ATSpectrograph CSC.
+    - See this :download:`configuration parameter override example file <_static/ATSpectrograph_example_config.yaml>` for the ATSpectrograph CSC.
+    - This file is the third configuration file to get loaded by the CSC and can override any previously declared values.
+    - These files are loaded using the `configurationToApply` attribute in the ``start`` command
+    - These are not expected to be required as part of regular operations and are meant to be used when a non-standard configuration is required
+    - If an override configuration file is also site specific, then a prefix should be added indicating which site it belongs with (e.g. ``summit_reduced_stage_travel.yaml``)
 
 #.  The labels file (e.g. ``_labels.yaml``) shall continue to exist with the same format and purpose.
-#.  Additional configuration files can provide new values for individual configuration parameters.
 
-    - See this :download:`example configuration file <_static/ATSpectrograph_example_config.yaml>` for the ATSpectrograph CSC.
+        - No label shall exist having the name ``default``
```

Comment:
What is this rule for? Other names to consider prohibiting:
* `init`, to avoid confusion with `_init.yaml`
* `<site>`, to avoid confusion with `_<site>.yaml`. Admittedly that's a bit risky since if we add a site we could add conflicts with existing labels. But such names seem unlikely. To enforce such a rule we would probably either prohibit any label that matches a file named `_<label>.yaml` or have a blessed set of site names.

I suggest checking enforcement in ts_config_* unit tests, rather than trying to enforce such rules in salobj's config loading code. I would like to keep salobj's loading code as simple as possible.

Reply:
I agree with all of this. Yes, the idea was to have an exclusion "list", but it should be kept small. Default is a bit of a special case since it somewhat occurs to people immediately but we want them to not use it since it's ambiguous.